jedv00-sp2

jedv00 opened the pull request:
Link to app: https://eso.vse.cz/~jedv00/sp2/

jedv00 committed:
jedv00-sp2

Link to app: https://eso.vse.cz/~jedv00/sp2/

nvbach91 commented on www/jedv00/sp2/search/index.html:
```diff
@@ -0,0 +1,131 @@
+                </div>
+                <div class="col-5 d-flex justify-content-end ">
+                    <div class="row mt-3 mb-3 justify-content-md-center">
+                    <button type="button" class="btn btn-primary" onclick="relocate()">📚</button>
```
bez inline javascript

jedv00 replied:
Mohu se jen zeptat proč? Jinak samozřejmě provedu :)

Děkuju

nvbach91 replied:
stejný důvod jako u CSS, jinak jsou tam faktory jako údržba, čitelnost, debugování, testování, performance - browser cache, aj. a hlavně to jde proti některým principům co jste se naučil/naučíte na jiných předmětech ohledně vývoje SW

nvbach91 commented on www/jedv00/sp2/index.html:
```diff
@@ -0,0 +1,149 @@
+        </div>
+        <script>
+            function relocate()
+            {
```
tohle jde do main.js

jedv00 committed:
v2

second and hopefully last push :)

nvbach91 commented on www/jedv00/sp2/js/script.js:
```diff
@@ -0,0 +1,304 @@
+  //Google Books API key
+  var apiKey = "AIzaSyCl_pxu4tYFtJ8O3QriNDfntlufgeRu6pE";
+  //Storage variable
+  var myStorage = window.localStorage;
```
je zbytecne davat do vlastni promenne

jedv00 replied:
> je zbytecne davat do vlastni promenne

proměnná zrušena

nvbach91 commented on www/jedv00/sp2/js/script.js:
```diff
@@ -0,0 +1,304 @@
+    return path[1];
+  }
+
+  var path = getPath();
```
to prechazeni mezi strankami nesmi byt udelane pres `location.href = ...` protoze se stahne jina stranka, stejne jako kdyby tam byl element `<a>` a pak to uz nebude single page aplikace

pro kazdou "view" udelejte jednu funkci, napr. renderSearchPage, renderViewPage, a budou renderovat DOM dynamicky bez presmerovani na jinou stranku

nebo udelejte dva sloupce do jedne stranky, vlevo bude search a vpravo bude view a budou se interagovat mezi sebou (pridam knihu vlevo a objevi se mi vpravo)

jedv00 replied:
> to prechazeni mezi strankami nesmi byt udelane pres `location.href = ...` protoze se stahne jina stranka, stejne jako kdyby tam byl element `<a>` a pak to uz nebude single page aplikace
> 
> pro kazdou "view" udelejte jednu funkci, napr. renderSearchPage, renderViewPage, a budou renderovat DOM dynamicky bez presmerovani na jinou stranku
> 
> nebo udelejte dva sloupce do jedne stranky, vlevo bude search a vpravo bude view a budou se interagovat mezi sebou (pridam knihu vlevo a objevi se mi vpravo)

předěláno na single page

nvbach91 commented on www/jedv00/sp2/js/script.js:
```diff
@@ -0,0 +1,304 @@
+  function search() {
+    //get value from input search
+    //trim function is for erasing spaces before and behind input
+    var searchValue = document.getElementById("searchInput").value.trim();
```
pouzijte jQuery + elementy vybrat predem, ne behem pouziti

jedv00 replied:
> pouzijte jQuery + elementy vybrat predem, ne behem pouziti

opraveno

nvbach91 commented on www/jedv00/sp2/js/script.js:
```diff
@@ -0,0 +1,304 @@
+    //get value from input search
+    //trim function is for erasing spaces before and behind input
+    var searchValue = document.getElementById("searchInput").value.trim();
+    var src =
```
pouzijte string interpolation

jedv00 replied:
> pouzijte string interpolation

opraveno všude

nvbach91 commented on www/jedv00/sp2/js/script.js:
```diff
@@ -0,0 +1,304 @@
+          var savedBooks = getStorage();
+          //remove table before input
+          src = "";
+          for (var i = 0; i < books.items.length; i++) {
```
lepsi forEach loop

jedv00 replied:
> lepsi forEach loop

upraveno na jquery each

nvbach91 commented on www/jedv00/sp2/js/script.js:
```diff
@@ -0,0 +1,304 @@
+            }
+
+            //add contents of books
+            src +=
```
string interpolation.

nvbach91 commented on www/jedv00/sp2/js/script.js:
```diff
@@ -0,0 +1,304 @@
+          for (var i = 0; i < books.items.length; i++) {
+            var item = books.items[i];
+            var year = getYearFromString(item.volumeInfo.publishedDate);
+            var ISBN = searchInIdentifiers(item.volumeInfo.industryIdentifiers);
```
`var isbn`

jedv00 replied:
> `var isbn`

opraveno

nvbach91 commented on www/jedv00/sp2/js/script.js:
```diff
@@ -0,0 +1,304 @@
+              "<td>" +
+              ISBN +
+              "</td>" +
+              "<td><button type='button' class='btn " + buttonColor + " buttonAdd' " + buttonDisabled + " data-toggle='modal' data-target='#exampleModal' data-id='" + item.id + "'>" + buttonText + "</button></td>" + 
```
tady je duvod proc pouzivat apostrofy a ne uvozovky

nvbach91 commented on www/jedv00/sp2/js/script.js:
```diff
@@ -0,0 +1,304 @@
+
+   //storage is just string.. i need to format it
+  function getStorage() {
+    return JSON.parse(myStorage.getItem("books")) || [];
```
tady je potreba pouzit try/catch

jedv00 replied:
> tady je potreba pouzit try/catch

přidáno

nvbach91 commented on www/jedv00/sp2/js/script.js:
```diff
@@ -0,0 +1,304 @@
+    myStorage.setItem("books", JSON.stringify(items));
+  }
+
+  function removeFromStorage(id) {
```
pro rychlejsi spravu polozek (vymaz, aktualizace) pouzijte objekt misto pole:
tj.:
```js
var books = {
  isbn1: {
    title: 'blabla',
  },
  isbn2: {
    title: 'bloblo',
  },
};

delete books['blabla']; // vymaz
books['bloblo'].title = 'new title'; // aktualizace
books['blublu'] = {  title: 'blublu', }; // pridavani
```

nvbach91 commented on www/jedv00/sp2/js/script.js:
```diff
@@ -0,0 +1,304 @@
+        //set storage
+        setStorage(oldItems);
+
+        //reaload search
```
nope.

co kdybyste misto toho reloadovani jenom zmenil attribut a styl tlacitka u knizky ktera byla pridana?

podobne jako u vymaz knizky z library, kde jste jenom smazal jedno `<tr>`?

jedv00 replied:
> nope.
> 
> co kdybyste misto toho reloadovani jenom zmenil attribut a styl tlacitka u knizky ktera byla pridana?
> 
> podobne jako u vymaz knizky z library, kde jste jenom smazal jedno `<tr>`?

změněno dle doporučení

nvbach91 reviewed (changes requested):
Ukoly:
- pridat thumbnail (vsude)
- pridat razeni abecedne podle nazvu, autora a published date
- pridat remove do search view
- pridat preview do popupu (title, obrazek, subtitle, description, publisher a dalsi info)

jedv00 committed:
finální úpravy

Opravy kódu okomentovány v předešlích komentářích.

Z úkolů provedeno:
- vytvořeno preview s knížkami (pouze pokud jsem ve vlastním seznamu a po kliknutí na title knížky - přijde mi to hezčí :) )
- přidáno tlačítko "Remove" do serch view

jedv00 commented:
> Ukoly:
> 
> * pridat thumbnail (vsude)
> * pridat razeni abecedne podle nazvu, autora a published date
> * pridat remove do search view - PŘIDÁNO
> * pridat preview do popupu (title, obrazek, subtitle, description, publisher a dalsi info) - edit ke komentu z commitu -> review je i v search view

jedv00 committed:
Delete search/index.html

Smazáno, protože je aplikace předělána na single page a toto již více není potřeba

nvbach91 reviewed (approved):
![how-to-create-presentation-slides-that-are-out-of-this-world-by-slidecomet-itseugenec-kaixinspeaking-28-638](https://user-images.githubusercontent.com/20724910/104008962-8b66ea80-51aa-11eb-9b47-d64347f9797a.jpg)
